test/Microsoft.VisualStudio.LanguageServices.Razor.Test/Editor/VisualStudioRazorParserTest.cs

```diff
@@ -578,26 +577,36 @@ private void RunTypeKeywordTest(string keyword)
             }
         }
 
-        private static void DoWithTimeoutIfNotDebugging(Func<int, bool> withTimeout)
+        private static void DoWithTimeoutIfNotDebugging(ManualResetEventSlim resetEvent)
         {
+            if (resetEvent.IsSet)
+            {
+                // Already fired due to a race; we'll bail early to avoid the wait.
+                return;
+            }
+
 #if DEBUG
             if (Debugger.IsAttached)
             {
-                withTimeout(Timeout.Infinite);
+                resetEvent.Wait(Timeout.Infinite);
             }
             else
             {
 #endif
-                Assert.True(withTimeout((int)TimeSpan.FromSeconds(1).TotalMilliseconds), "Timeout expired!");
+                // Possible race between reset event firing and this wait. We'll only fail after the Wait
+                // if the reset event is not set.
+
+                resetEvent.Wait(TimeSpan.FromSeconds(1));
```

Comment:
`Assert.True(resetEvent.Wait(TimeSpan.FromSeconds(1)), "Timeout expired");`

Reply:
That's not the same logic